database/bulkcreate.go

```diff
@@ -49,9 +50,12 @@ func (api *bulkCreate[T]) Commit() (sql.Result, errors.TracerError) {
 			api.pending[0].Meta().WriteColumns(),
 			api.pending[0].Meta().PrimaryKey(),
 		)
-		query     = qb.Insert(writeCols...)
-		stmt, err = query.ParameterizedSQL()
+		query = qb.Insert(writeCols...)
 	)
+	if api.upsert {
+		query.OnDuplicate(api.pending[0].Meta().WriteColumns())
+	}
+	stmt, err := query.ParameterizedSQL()
```

Comment:
minor nit, but if we have a var block we should declare all the vars there and not use assign and declare syntax as well (within the same scope)